packages/passport-interface/src/PassportPopup.ts

```diff
@@ -45,8 +45,11 @@ export function usePassportPopupSetup() {
       return;
     }
 
-    const search = window.location.search;
-    const params = new URLSearchParams(search);
+    // Single page client-side apps need a hash router because
+    // there is no server to direct all traffic.
+    // If the URL contains a hash it will be removed to read the search parameters.
+    const url = new URL(window.location.href.replace("#", ""));
+    const params = url.searchParams;
```

Comment:
this comment is a little difficult for me to understand, and i'm not sure that this solution is robust

Reply:
Please let me know if it's clearer now. The check should only catch cases where hash routing is used now (no anchors etc).